packages/amplify-graphql-relational-transformer/src/graphql-has-one-transformer.ts

```diff
@@ -51,6 +69,50 @@ export class HasOneTransformer extends TransformerPluginBase {
     this.directiveList.push(args);
   };
 
+  /** During the preProcess step, modify the document node and return it
+   * so that it represents any schema modifications the plugin needs
+   */
+  mutateSchema = (context: TransformerPreProcessContextProvider): DocumentNode => {
+    const document: DocumentNode = produce(context.inputDocument, draftDoc => {
+      draftDoc.definitions.forEach(def => {
+        if (def.kind === 'ObjectTypeDefinition' || def.kind === 'ObjectTypeExtension') {
+          def?.fields?.forEach(field => {
+            field?.directives?.forEach(dir => {
+              if (dir.name.value === directiveName) {
+                const connectionAttributeName = getConnectionAttributeName(def.name.value, field.name.value);
+                let hasFieldsDefined = false;
+                let removalIndex = -1;
+                dir?.arguments?.forEach((arg, idx) => {
+                  if (arg.name.value === 'fields') {
+                    if ((arg.value.kind === 'StringValue' && arg.value.value) || (arg.value.kind === 'ListValue' && arg.value.values && arg.value.values.length > 0)) {
+                      hasFieldsDefined = true;
+                    } else {
+                      removalIndex = idx;
+                    }
+                  }
+                });
+                if (removalIndex !== -1) {
+                  dir?.arguments?.splice(removalIndex, 1);
+                }
+                if (!hasFieldsDefined) {
+                  // eslint-disable-next-line no-param-reassign
+                  dir.arguments = [makeArgument('fields', makeValueNode(connectionAttributeName)) as WritableDraft<ArgumentNode>];
+                  def?.fields?.push(
+                    makeField(
+                      connectionAttributeName, [], isNonNullType(field.type) ?
+                        makeNonNullType(makeNamedType('ID')) : makeNamedType('ID'), [],
+                    ) as WritableDraft<FieldDefinitionNode>,
+                  );
+                }
+              }
+            });
```

Comment:
Same as the previous comment. Please see if we can use filter to simply the logic.